src/App.js

```diff
@@ -1,43 +1,54 @@
 import React, {useEffect, useState} from 'react';
-import AddTodoForm from './AddTodoForm';
 import TodoList from './TodoList';
+import { BrowserRouter, Routes, Route } from 'react-router-dom';
+import FavoritesPage from './FavePage/FavoritesPage';
+import Header from './Header';
 
 
-// Function to preserve list upon refresh. Works with local storage.
+// Function to preserve list upon refresh. Works with local storage.--sb
 
 const useSemiPersistentState = () => {
-
+  
   const [todoList, setTodoList] = useState(
     JSON.parse(localStorage.getItem("savedTodoList")) || []
   );
+
   useEffect(() => {
     localStorage.setItem("savedTodoList", JSON.stringify(todoList));
   }, [todoList]);
+
   return [todoList, setTodoList];
 };
 
 
 
 function App() {
-
-  const [todoList, setTodoList] = useSemiPersistentState();
-
-  const addTodo = (newTodo) => {
-    setTodoList([...todoList, newTodo])
-  };
-
-  const removeTodo = (id) => {
-    const newTodoList = todoList.filter((todo) =>
-      id !== todo.id);
-    setTodoList(newTodoList)
-  }; 
-
   return (
-    <>
-      <h1>Title in App.js</h1>
-      <AddTodoForm onAddTodo={addTodo} />
-      <TodoList todoList={todoList} onRemoveTodo={removeTodo} />
-    </>
-  );
+      <BrowserRouter>
+        <Routes>
+        
+        {/* home */}
+          <Route path="/" element={
+  <>        
+            <Header />
+            <h1>Todo List Title</h1>
+            <TodoList useSemiPersistentState={useSemiPersistentState}/>
+  </>     
+        }></Route>
+        
+        
+        {/* FavePage */}
+          <Route path="/Favorites" element={
+  <>
+            <Header />
+            <h1>Favorites</h1>
+            <FavoritesPage useSemiPersistentState={useSemiPersistentState} />
+  </>
+        }></Route>
+        
+        </Routes>
```

Comment:
here we are repeating Header, but what we can do is just  use Header and Footer outside routes, so they will be in all pages. anther things is it would be if you <h1></h> in favorites page and todolist page instead of putting them inside <></>. just like this : ` <Router>
    <Header/>
      <Route exact path="/" component={Main} />
      <Route exact path="/home" component={Main} />
      <Route exact path="/about" component={AboutContainer} />
      <Route exact path="/blog" component={BlogContainer} />
    <Footer/>
  </Router>`

Reply:
this way, it more readable for everyone.

Reply:
Thank you! I'll make those changes now.

Reply:
Done! I followed the example you shared and it looks cleaner.